oak-core/src/test/java/org/apache/jackrabbit/oak/security/user/UserAuthenticationTest.java

```diff
@@ -187,6 +189,29 @@ public void testAuthenticateImpersonationCredentials2() throws Exception {
         assertTrue(authentication.authenticate(new ImpersonationCredentials(sc, mockAuthInfo(userId))));
     }
 
+    @Test
+    public void testImpersonationByAdministrator() throws Exception {
```

Comment:
once you have refactored the patch, please add a bit of testing to ImpersonationImplTest that essentially verifies that group membership is not being evaluated during thee 'impersonation.allows' call :)

something like:
- get the Impersonation object from user
- call allows(Subject) with a subject that you have created 
> test1: configured-can-impersonate-all principal name not contained in the principal-set => not allowed
> test2: configured-can-impersonate-all principal name is contained in the principal set of the subject => must be allowed.
> more tests depending on th impl: either verify that type of principal does not matter.... or if you prefer that this option is limited to GroupPrincipals -> add a test that verifies that the config option is being ignored if no group-principal with the configured name is present

that's how i believe it should work :)